pkg/controller/controller.go

```diff
@@ -522,3 +523,39 @@ func isInstanceReady(instance *v1alpha1.Instance) bool {
 
 	return false
 }
+
+// IsOwnedBy checks if the given object has a controller ownerReference set to the given owner
+func IsOwnedBy(obj metav1.Object, owner metav1.Object) bool {
+	ref := GetControllerOf(obj)
+	if ref == nil {
+		return false
+	}
+	return ref.UID == owner.GetUID()
+}
+
+// GetControllerOf returns the controllerRef if controllee has a controller,
```

Comment:
Yes, we should export the helper functions. There are two options:
* move them next to the meta/v1 package in k8s.io/apimachinery, as you suggested
* move them to kubernetes-util (yet to be created, cc @mengqiy)

I prefer moving them to k8s.io/apimachinery, but i'd like to hear others' opinions. cc @deads2k @sttts